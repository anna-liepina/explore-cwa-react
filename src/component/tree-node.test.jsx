import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import TreeNode from './tree-node';

describe('<TreeNode/>', () => {
    const props = {
        text: 'props.text',
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        chunks: [
            { v: 't', isMatch: true },
            { v: 'e', isMatch: false },
            { v: 'xt' },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<TreeNode {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<TreeNode {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });

        describe('prop combinations', () => {
            [
                [
                    'collapsed with nodes',
                    {
                        isExpanded: false,
                        nodes: [{ text: 'nodes[0].text' }],
                    },
                ],
                [
                    'expanded with NO nodes',
                    {
                        isExpanded: true,
                        nodes: [],
                    },
                ],
                [
                    'expanded with "hidden" nodes',
                    {
                        isExpanded: true,
                        nodes: [{ text: 'nodes[0].text' }],
                    },
                ],
                [
                    'expanded with "visible" nodes',
                    {
                        isExpanded: true,
                        nodes: [{ text: 'nodes[0].text', isVisible: true }],
                    },
                ],
            ].forEach(([desc, propsCombination]) => {
                it(desc, () => {
                    const { asFragment } = render(<TreeNode {...props} {...propsCombination} />);

                    expect(asFragment()).toMatchSnapshot();
                });
            });
        });
    });
});
