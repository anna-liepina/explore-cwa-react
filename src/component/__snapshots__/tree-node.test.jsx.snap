// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeNode/> render prop combinations collapsed with nodes 1`] = `
<DocumentFragment>
  <div
    class="tree-node--with-children "
    data-cy=""
  >
    <span>
      props.text
    </span>
  </div>
</DocumentFragment>
`;

exports[`<TreeNode/> render prop combinations expanded with "hidden" nodes 1`] = `
<DocumentFragment>
  <div
    class="tree-node--with-children-expanded "
    data-cy=""
  >
    <span>
      props.text
    </span>
  </div>
</DocumentFragment>
`;

exports[`<TreeNode/> render prop combinations expanded with "visible" nodes 1`] = `
<DocumentFragment>
  <div
    class="tree-node--with-children-expanded "
    data-cy=""
  >
    <span>
      props.text
    </span>
    <div
      class="tree-node "
      data-cy="-0"
      data-node="0"
    >
      <span>
        nodes[0].text
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TreeNode/> render prop combinations expanded with NO nodes 1`] = `
<DocumentFragment>
  <div
    class="tree-node--with-children-expanded "
    data-cy=""
  >
    <span>
      props.text
    </span>
  </div>
</DocumentFragment>
`;

exports[`<TreeNode/> render with default/required props 1`] = `
<DocumentFragment>
  <div
    class="tree-node "
    data-cy=""
    data-node="0"
  >
    <span>
      props.text
    </span>
  </div>
</DocumentFragment>
`;

exports[`<TreeNode/> render with optional/required props 1`] = `
<DocumentFragment>
  <div
    class="tree-node optProps.className"
    data-cy="optProps.data-cy"
    data-node="0"
  >
    <span
      class="tree-node--pattern-match"
    >
      t
    </span>
    <span>
      e
    </span>
    <span>
      xt
    </span>
  </div>
</DocumentFragment>
`;
